src/rpm_ostree/actor.rs

```diff
@@ -56,51 +59,91 @@ impl Handler<StageDeployment> for RpmOstreeClient {
+                    //     rebased to a version tag. Here deploy would work but would lead to a weird UX:
+                    //     rpm-ostree status would show the version tag in the origin after we moved on to
+                    //     another version.
+                    // 3 - The image name we are following has changed (new registry, new name)
+                    //     In that case `deploy` won't work and we need to rebase to the new refspec.
+
+                    // case 1
+                    // Migrating from OSTree to OCI, signaled by the fake checksum
+                    let rebase_target = if local_deploy.get_checksum() == *OCI_MIGRATION_CHECKSUM_SIGNAL
+                        // case 2: the booted tag does not match the stream
+                        || booted_oci_ref.tag() != local_deploy.get_fcos_update_stream().ok().as_deref()
+                        // case 3: new image location
+                        || booted_oci_ref.registry() != target_pullspec.registry()
+                        || booted_oci_ref.repository() != target_pullspec.repository()
+                    {
+                        // Overwrite the tag with the stream we are targetting because if it's a version tag
+                        // sticking it to the new deployment would be incorrect.
+                        // Also, if the node previously rebased to a digested ostree img ref, we want to
+                        // set the tag with the stream value as we are actually picking up that stream.
+                        let tagged_rebase_ref = Reference::with_tag(
+                            target_pullspec.registry().to_string(),
+                            target_pullspec.repository().to_string(),
+                            local_deploy.get_fcos_update_stream()?,
+                        );
+                        // craft a new ostree imgref object with the tagged oci reference we'll use for
+                        // the rebase command
+                        let rebase_target = OstreeImageReference {
+                            sigverify: booted_imgref.sigverify,
+                            imgref: ostree_ext::container::ImageReference {
+                                transport: booted_imgref.imgref.transport,
+                                name: tagged_rebase_ref.whole(),
+                            },
+                        };
+                        Some(rebase_target)
+                    } else {
+                        None
+                    };
+
+                    let options = StagingOptions {
+                        allow_downgrade: msg.allow_downgrade,
+                        custom_origin,
+                        rebase: rebase_target,
+                    };
+
+                    trace!("request to stage release: {:?}", &msg.release);
+                    super::cli_deploy::deploy_locked(msg.release, options)
```

Comment:
I guess we could factor this out of the match statement now and have the match be purely about constructing `options`.

```rust
let options = match &msg.release.payload { ... };
trace!("request to stage release: {:?}", &msg.release);
super::cli_deploy::deploy_locked(msg.release, options) // last statement
```

Reply:
i ended up nuking the options structs since I removed the custom origin code. 
The match now is just about creating an `Option<OSTreeImageReference>` to rebase to